proofs/alf/cvc5/Cvc5.smt3

```diff
@@ -78,13 +84,27 @@
       ((run_evaluate (div x y))          (alf.zdiv (run_evaluate x) (run_evaluate y)))
       ((run_evaluate (div_total x y))    (let ((d (run_evaluate y)))
                                            (alf.ite (alf.is_eq d 0) 0 (alf.zdiv (run_evaluate x) d))))
+      ((run_evaluate (mod x y))          (alf.zmod (run_evaluate x) (run_evaluate y)))
+      ((run_evaluate (to_real x))        (alf.to_q (run_evaluate x)))
+      ((run_evaluate (to_int x))         (alf.to_z (run_evaluate x)))
+      ((run_evaluate (is_int x))         (alf.is_eq (alf.to_q (alf.to_z x)) (alf.to_q x)))
+
 
       ; strings
       ((run_evaluate (str.++ xs yss))    (alf.concat (run_evaluate xs) (run_evaluate yss)))
       ((run_evaluate (str.len x))        (alf.len (run_evaluate x)))
       ((run_evaluate (str.substr x n m)) (let ((r (run_evaluate n)))
                                             (alf.extract (run_evaluate x) r (alf.add r (run_evaluate m) -1))))
       ((run_evaluate (str.contains x y)) (alf.not (alf.is_neg (alf.find (run_evaluate x) (run_evaluate y)))))
+      ((run_evaluate (str.replace x y z))(let ((ex (run_evaluate x)))
+                                         (let ((ey (run_evaluate y)))
+                                         (let ((i (alf.find ex ey)))
+                                            (alf.ite (alf.or (alf.is_eq i 0) (alf.is_neg (alf.neg i)))
+                                              (alf.concat (alf.extract ex 0 (alf.add i -1)) (run_evaluate z) (alf.extract ex (alf.add i (alf.len ey)) (alf.len ex)))
+                                              ex)))))
+      ((run_evaluate (str.indexof x y n))(let ((ex (run_evaluate (str.substr x n (str.len x)))))
+                                         (let ((ey (run_evaluate y)))
+                                            (alf.find (alf.to_str ex) (alf.to_str ey)))))
```

Comment:
Did you forget to add `n`? Also, why is there a need to `alf.to_str`? It's not used in the above case...
```suggestion
                                            (alf.add n (alf.find (alf.to_str ex) (alf.to_str ey))))))
```

Reply:
Yes good catch. I'll also add an SMT-LIB example where this failed.

Reply:
We should use `alf.to_str` since `alf.find` is overloaded. If it was the case that `ex` was a concatenation term, then `alf.find` would return the *child index* of where `ey` occurs. By casting to strings, we ensure `alf.find` does not evaluate if these are not string literals.

Reply:
More hacks 😁!!!

Reply:
Haha, yes. Having only basic evaluation operators in the core language requires hacks, although I agree these hacks should be encapsulated in utilities.